src/coreclr/jit/hwintrinsic.cpp

```diff
@@ -778,7 +778,11 @@ GenTree* Compiler::getArgForHWIntrinsic(var_types            argType,
             {
                 arg = impSIMDPopStack();
             }
+#if defined(FEATURE_MASKED_HW_INTRINSICS)
+            assert(varTypeIsSIMD(arg) || varTypeIsMask(arg));
```

Comment:
There's an argument to be made that `varTypeIsSIMD()` should return true for `TYP_MASK`, but I didn't want to go that far, as I suspect it will break something.

Reply:
....instead I've added `varTypeIsSIMDOrMask()`

Reply:
In this case I think it's very desirable that `varTypeIsSIMD` does not return true for `TYP_MASK`, so I think having a `varTypeIsSimdOrMask` is the right approach.

`TYP_MASK` is a different register set and has largely different semantics/considerations processing wise, even if it is only ever used with SIMD code.